packages/metering/lib/processors/customer-tracking.ts

```diff
@@ -0,0 +1,83 @@
+import db from '@nangohq/database';
+import { Subscriber } from '@nangohq/pubsub';
+import { accountService, connectionService, environmentService, productTracking } from '@nangohq/shared';
+import { Err, Ok, report, stringifyError } from '@nangohq/utils';
+
+import { logger } from '../utils.js';
+
+import type { Transport, UsageEvent } from '@nangohq/pubsub';
+import type { Result } from '@nangohq/utils';
+
+export class CustomerTracking {
+    private subscriber: Subscriber;
+
+    constructor(transport: Transport) {
+        this.subscriber = new Subscriber(transport);
+    }
+
+    public start(): void {
+        logger.info('Starting customer tracking subscriber...');
+
+        this.subscriber.subscribe({
+            consumerGroup: 'customer-tracking',
+            subject: 'usage',
+            callback: async (event) => {
+                logger.info(`Processing customer tracking event`, { event });
+                const result = await process(event);
+                if (result.isErr()) {
+                    report(new Error(`Failed to process customer tracking event: ${result.error}`), { event });
+                    return;
+                }
+            }
+        });
+    }
+}
+
+async function process(event: UsageEvent): Promise<Result<void>> {
+    try {
+        switch (event.type) {
+            case 'usage.connections': {
+                logger.info(`Tracking usage for account ${event.payload.properties.accountId}`);
+                if (typeof event.payload.properties['environmentId'] === 'number') {
+                    await notifyOnProdUsageThreshold({
+                        accountId: event.payload.properties.accountId,
+                        environmentId: event.payload.properties['environmentId']
+                    });
+                }
+                // No action for tracking connections
+                return Ok(undefined);
+            }
+            default:
+                return Err(`Unknown customer tracking event type: ${event.type}`);
```

Comment:
It is not really an error, here we are deliberately not interested in all the other types. In this case I would return silently (ie: `Ok(undefined)`)